linera-execution/src/wasm/runtime_actor/requests.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Different request types for different runtimes.
+
+use super::responses::{SyncOrAsyncResponse, SyncResponseSender};
+use crate::{CallResult, UserApplicationId};
+use futures::channel::oneshot;
+use linera_base::{
+    data_types::{Amount, Timestamp},
+    identifiers::{ChainId, SessionId},
+};
+use linera_views::batch::Batch;
+use std::fmt::{self, Debug, Formatter};
+
+/// Requests shared by contracts and services.
+pub enum BaseRequest {
+    /// Requests the current chain id.
+    ChainId {
+        response: SyncResponseSender<ChainId>,
```

Comment:
nit: Here and all the places, `response_sender` would be the expected (and more correct) name.

Reply:
Done.